VMDK Import crashes when no Snapshot is present in VM object

<!-- This form is for bug reports and feature requests ONLY!
Also make sure that you visit our User Guide at https://kubevirt.io/user-guide/
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
 /kind bug
> /kind enhancement


**What happened**:
When trying to import a VMDK from a vcenter instance the importer container crashes with a SIGSEGV in the vddk-datasource.go file.
**What you expected to happen**:
Succesful import of the VMDK File
**How to reproduce it (as minimally and precisely as possible)**:
Import a VM from vcenter (using source: vddk in the yaml) which never had a snapshot taken. 
However simply deleting all snapshots the VM still seems to import fine.
**Anything else we need to know?**:
I think this might go back to the VM Object not having a snapshot list property when no snapshot was ever taken before. 
**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`): v1.28.3
- Kubernetes version (use `kubectl version`): 1.19.4
- Cloud provider or hardware configuration: 4 Node vSphere Cluster
- Install tools:
- Others:

Thanks for your debugging efforts. I think I have managed to reproduce this, but only when the "backingFile" field in my yaml does not exactly match the VMDK name:

```
YAML backingFile: "[iSCSI_Datastore] marnold-nosnap/marnold-nosnap.vmdk "
vCenter Virtual Disk: "[iSCSI_Datastore] marnold-nosnap/marnold-nosnap.vmdk"

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x208d6d4]

goroutine 1 [running]:
kubevirt.io/containerized-data-importer/pkg/importer.(*VMwareClient).FindDiskFromName(0xc000422480, 0xc000042066, 0x35, 0x1b, 0xc000044174, 0x8)
	pkg/importer/vddk-datasource.go:398 +0x424
```

Does this match the crash you are seeing? I found a few places in the code that can fail in similar ways, so I want to make sure I get the right one.

Hi, Unfortunately I ran into this issue again but taking a Snapshot did not resolve it. 
I am 100% certain about the backingFile name in the YAML:
`backingFile: "[vsanDatastore] 07de1b60-1683-085c-2706-7cd30a57dc28/ubuntu-container.vmdk" `
Copied from vCenter: `[vsanDatastore] 07de1b60-1683-085c-2706-7cd30a57dc28/ubuntu-container.vmdk`
And I keep seeing Go Exceptions in the importer again:
`I0204 13:17:23.968279       1 importer.go:52] Starting importer
I0204 13:17:23.968395       1 importer.go:132] begin import process
I0204 13:17:23.968409       1 vddk-datasource.go:763] Creating VDDK data source: backingFile [[vsanDatastore] 07de1b60-1683-085c-2706-7cd30a57dc28/ubuntu-container.vmdk], currentCheckpoint [], previousCheckpoint [], finalCheckpoint []
I0204 13:17:24.058113       1 vddk-datasource.go:447] VM 421f2fc0-afd7-c61a-be03-581e96f297c8 found in datacenter Datacenter:datacenter-3 @ /Datacenter: vm-3913
I0204 13:17:24.059951       1 vddk-datasource.go:308] Current VM power state: poweredOff
I0204 13:17:24.063749       1 vddk-datasource.go:329] Logged out of VMware.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x2089b84]
goroutine 1 [running]:
kubevirt.io/containerized-data-importer/pkg/importer.(*VMwareClient).FindDiskFromName(0xc0002f0e80, 0xc000044016, 0x4a, 0x1b, 0xc000042114, 0xc)
	pkg/importer/vddk-datasource.go:397 +0x424
kubevirt.io/containerized-data-importer/pkg/importer.createVddkDataSource(0xc000046012, 0x16, 0xc000048057, 0x1b, 0xc000042114, 0xc, 0xc000040064, 0x3b, 0xc00004800e, 0x24, ...)
	pkg/importer/vddk-datasource.go:778 +0x429
kubevirt.io/containerized-data-importer/pkg/importer.NewVDDKDataSource(...)
	pkg/importer/vddk-datasource.go:759
main.main()
	cmd/cdi-importer/importer.go:168 +0x1c76`

--- a/pkg/importer/vddk-datasource.go
+++ b/pkg/importer/vddk-datasource.go
@@ -394,10 +394,14 @@ func (vmware *VMwareClient) FindDiskFromName(fileName string) (*types.VirtualDis
 		klog.Errorf("Unable to list snapshots: %s\n", err)
 		return nil, err
 	}
-	if disk := vmware.FindDiskInSnapshotTree(snapshot.Snapshot.RootSnapshotList, fileName); disk != nil {
-		return disk, nil
+	if snapshot.Snapshot == nil {
+		klog.Errorf("No snapshots on this virtual machine.")
+	} else {
+		if disk := vmware.FindDiskInSnapshotTree(snapshot.Snapshot.RootSnapshotList, fileName); disk != nil {
+			return disk, nil
+		}
 	}
-	return nil, errors.New("could not find target VMware disk")
+	return nil, fmt.Errorf("disk '%s' is not present in VM hardware config or snapshot list", fileName)
 }
 
 // FindSnapshotDiskName finds the name of the given disk at the time the snapshot was taken

--- a/pkg/importer/vddk-datasource_test.go
+++ b/pkg/importer/vddk-datasource_test.go
@@ -300,6 +300,26 @@ var _ = Describe("VDDK data source", func() {
 			Expect(extent.Length).To(Equal(source.ChangedBlocks.ChangedArea[index].Length))
 		}
 	})
+
+	It("should not crash when the disk is not found and has no snapshots", func() {
+		newVddkDataSource = createVddkDataSource
+		diskName := "testdisk.vmdk"
+		wrongDiskName := "wrong.vmdk"
+
+		currentVMwareFunctions.Properties = func(ctx context.Context, ref types.ManagedObjectReference, property []string, result interface{}) error {
+			switch out := result.(type) {
+			case *mo.VirtualMachine:
+				if property[0] == "config.hardware.device" {
+					out.Config = createVirtualDiskConfig(wrongDiskName, 12345)
+				}
+			}
+			return nil
+		}
+
+		_, err := NewVDDKDataSource("http://vcenter.test", "user", "pass", "aa:bb:cc:dd", "1-2-3-4", diskName, "", "", "false", v1.PersistentVolumeFilesystem)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(Equal("disk 'testdisk.vmdk' is not present in VM hardware config or snapshot list"))
+	})
 })
 
 type mockNbdOperations struct{}

